python/ray/data/tests/test_numpy_support.py

```diff
@@ -27,22 +28,31 @@ def assert_structure_equals(a, b):
     assert a.dtype == b.dtype
     assert a.shape == b.shape
     for i in range(len(a)):
-        assert np.array_equiv(a[i], b[i]), (i, a, b)
+        assert np.array_equal(a[i], b[i]), (i, a[i], b[i])
 
 
-def test_list_of_scalars(ray_start_regular_shared):
+def test_list_of_scalars(ray_start_regular_shared, restore_data_context):
+    # Disable (automatic) fallback to `ArrowPythonObjectType` extension type
+    DataContext.get_current().enable_fallback_to_arrow_object_ext_type = False
```

Comment:
Why do we need to disable fallback here?

Reply:
Previously quite a few of these tests were actually invoking the fallback that shouldn't have happened, hence i'm making these tests stricter to not use the fallback (unless testing the fallback explicitly)